### ISSUE TITLE
OmniSharp Roslyn analyzers stop analyzing after initial display

### ISSUE THREAD

**Issue by NoTuxNoBux** [wackoisgod/com.unity.ide.vscode#5]
Hi

(Not sure if this is specifically an issue caused by the code in this package, but I wanted to post this here anyway to get some input and help pinpoint where this issue lies, and it's probably relevant to others trying to migrate.)

We've been using the new SDK-style projects from this package for a while, and it works as we would expect. However, specifically on Windows with OmniSharp, configured Roslyn analyzers seem to kick in once, and then stop working entirely. This literally means you open a file in VSCode with OmniSharp, all your errors and warnings show up correctly; you save once, and they're gone.

It's particularly strange since:

* OmniSharp doesn't seem to report any issues.
* We verified they are configured correctly and they _do_ kick in, suggesting this package did everything right and OmniSharp picks them up and uses them. Also, reverting to the old package works fine, suggesting the Roslyn analyzers are not at fault either.
* It _only_ happens on Windows. I use OmniSharp on Linux myself, and it doesn't have this issue, even with the same .NET SDK versions (6 and 7).

I can't test Linux with the old package since legacy-style projects require .NET Framework, which is Windows-only.

**Comment by NoTuxNoBux** [comment]
I think I figured it out. The `Compile` items have to be **relative** to the project folder. Currently they are absolute, like this:

```xml
<Compile Include="C:\Users\Me\unity-project\Assets\Scripts\Foo.cs" />
```

If I change them all to be like this:

```xml
<Compile Include="Assets\Scripts\Foo.cs" />
```

I can reproducibly fix the issue.

I can also reproduce this outside Unity projects separate from this package: the moment you use full paths, analyzers stop linting after the first time. Must be something internal in OmniSharp.

The MSBuild documentation doesn't appear to be entirely clear on whether absolute paths are even allowed here, but since the rest of autocompletion works, I'm guessing it should, but it's a bug that probably won't get much attention right now since most modern .NET projects don't use explicit compile items like this package and the deprecated Unity package do.

**Comment by NoTuxNoBux** [comment]
I reported an issue upstream here: https://github.com/OmniSharp/omnisharp-roslyn/issues/2506

Generating relative paths here is a workaround that should also fix it, though.

**Comment by NoTuxNoBux** [comment]
Hmm, I think there may be a bug in this package either way. The [code does `EscapedRelativePathFor` for each of the `Compile` items](https://github.com/wackoisgod/com.unity.ide.vscode/blob/8f99762018e5b1e38907d2eb394c323469a75822/Packages/com.unity.ide.vscode/Editor/ProjectGeneration/ProjectGeneration.cs#L475), but they are never made relative in practice. [The same happens for the `None` items](https://github.com/wackoisgod/com.unity.ide.vscode/blob/8f99762018e5b1e38907d2eb394c323469a75822/Packages/com.unity.ide.vscode/Editor/ProjectGeneration/ProjectGeneration.cs#L410), but those _do_ work.

**Comment by NoTuxNoBux** [comment]
FWIW it turns out I was on commit 67c02d0abf348966163dac835afdfcd630d113a6 and that doesn't include the above line. The latest, 8f99762018e5b1e38907d2eb394c323469a75822, does, but completely breaks the plugin for me and generates old-style .NET Framework projects again.

EDIT: Ignore that, I got confused and need to be using the sdk-style-support branch. The actual latest commit is 260436f0a018a5816cbf054f32ca82b795fa9227, but it breaks due to another reason unrelated to this one (see https://github.com/wackoisgod/com.unity.ide.vscode/issues/6).

**Comment by NoTuxNoBux** [comment]
Turns out that this is already fixed in aa45c826e24a0815c046d0a6b715ff76b1f8a624, ugh. Well, at least I discovered a bug in OmniSharp.
